server/src/main/java/io/crate/execution/dml/TranslogIndexer.java

```diff
@@ -113,9 +139,13 @@ private IndexDocumentBuilder populateLuceneFields(BytesReference source) throws
             Object castValue = valueForInsert(indexer.dataType, entry.getValue());
             if (castValue != null) {
                 indexer.valueIndexer.indexValue(castValue, docBuilder);
-                if (tableIndexSources.containsKey(column)) {
-                    addIndexField(docBuilder, tableIndexSources.get(column), castValue);
```

Comment:
A much simpler fix that wouldn't touch INdexer at all would be using indexer.dataType and handle ObjectType separately, ie try to use `innerTypes` if `tableIndexSources.containsKey(column)` returns false. 

But I wanted to keep Indexer and TranslogINdexer in sync, especially after seeing that Indexer handles some tricky cases like
` if (value instanceof Iterable<?> it) {`